pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -510,7 +510,7 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	if err := resources.ValidateParamArrayIndex(ctx, pipelineSpec, pr); err != nil {
 		// This Run has failed, so we need to mark it as failed and stop reconciling it
 		pr.Status.MarkFailed(ReasonObjectParameterMissKeys,
-			"PipelineRun %s/%s parameters is missing object keys required by Pipeline %s/%s's parameters: %s",
+			"PipelineRun %s/%s array indexing params fail validation by Pipeline %s/%s's parameters: %s",
```

Comment:
the reason `ReasonObjectParameterMissKeys` (`ObjectParameterMissKeys`) does not align with the message

Reply:
Oh thanks!

Reply:
I will also add test for this